### ISSUE TITLE
Tenant Image Pull Secrets

### ISSUE THREAD

**Issue by dforbes-pivotal** [minio/operator#611]
**Is your feature request related to a problem? Please describe.**
Trying to use a private (air gapped) image repository to create a minio tenant instance.  The configuration allows me to specify the private repository easily enough but my private repo requires authentication and the configuration in the spec for the imagePullSecrets does not seem to be honoured.  

To add to the behaviour I observed, by pre-loading the minio image on each worker node the system happily started up the minio service as the usage was set to only pull if needed.  Unfortunately the console seems to be hard coded to always pull and without the authentication to the repository this means that it never started the tenant's console.  Thus another potential issue that the configuration to make it pull if required is not honoured either.

**Describe the solution you'd like**
Add to the specification for the tenant yaml to include imagePullSecrets.

**Describe alternatives you've considered**
As a workaround I created a separate daemonset to pull down the minio image on all nodes and thus the minio service itself started.  Did not work for the console. 

**Additional context**
The documentation is a little light in explaining how to configure things, some more worked examples might help me.
Also - on a slightly different note, I tried to install using krew but this seems to mandate a connection to github which was impossible in my air-gapped environment.

**Comment by ethernoy** [comment]
I found that the CRD of tenants does not match with the sample value provided in helm values.

## CRD
```yaml
imagePullSecret:
    properties:
      name:
        type: string
```

## values.yaml

``` yaml
tenants:
  # Tenant name
  - name: minio1
    namespace:
    ## Registry location and Tag to download MinIO Server image
    image:
      repository: minio/minio
      tag: RELEASE.2021-04-06T23-11-00Z
      pullPolicy: IfNotPresent
    imagePullSecrets: []
```

imagePullSecrets (note the 's') is an array in tenant definition of the helm chart, but imagePullSecret is an object in tenant CRD.

**Comment by alidadar7676** [comment]
By the way, The correct definition of `imagePullSecrets`, based on the spec of Pod in Kubernetes, should be a list of `LocalObjectReference`s, not a single `LocalObjectRefrence` value. So the `values.yaml` has the correct value for `imagePullSecrets` and the `tenant` CRD should change.

**Comment by harshavardhana** [comment]
Fixed in fc3e26247cd574fb82af39e1be4b67663c9268a5 - we will update the CRD eventually to support an array for now its a single value.
